### DISCUSSION TITLE
Error connect to sqlite3 - SQLite does not allow the charset to be changed in the DSN

### DISCUSSION THREAD

**Discussion by vlad2005** [phalcon/cphalcon#15377] [Q&A] [answered]
Hi!
I'm new to phalcon framework, so maybe i do something.
I have configured an project to use Phalcon with sqlite3 database, but when i try to access sqlite database i got this error to nginx log:
`875: *121 FastCGI sent in stderr: "PHP message: PHP Notice:  SQLite does not allow the charset to be changed in the DSN. in /home/.../app/config/services.php on line 82" while reading response header from upstream, client: 127.0.0.1, server: default, request: "GET /test/register HTTP/1.1", upstream: "fastcgi://unix:/var/run/php/php7.4-fpm.sock:", host: "localhost:8000"
`
When i test vanilla php code with sqlite3 everything work, so is something related to Phalcon.

I use Phalcon 4.1 with php 7.4 on Ubuntu 18.04
Thanks in advance!

**Comment by quasipickle** [comment, marked as answer]
It's not the config file that's relevant, but services.php, line 82.

Phalcon uses PDO, so that's an error being generated by PDO.  From some quick searching, it appears SQLite doesn't really have a charset setting.

**Reply by vlad2005** [reply]
Thanks!
I will remove config file from my initial post.
So what should I do to be able to use sqlite with Phalcon framework?

**Reply by quasipickle** [reply]
You can use SQLite, just don’t set the charset.

**Reply by vlad2005** [reply]
I do not explicitly set charset anywhere. Everything is created using devtools.
The sqlite database is created with the default charset utf-8.

**Reply by vlad2005** [reply]
I finally solved it by modifying part of the services.php file
`$di->setShared('db', function () {
    $config = $this->getConfig();

    $class = 'Phalcon\Db\Adapter\Pdo\\' . $config->database->adapter;

    //specific to Sqlite
    $params = [
        'dbname'   => $config->database->dbname
    ];
    
    return new $class($params);
});`

For some reason, leaving all parameters in $params, except charset, not working. Only need dbname.

**Reply by quasipickle** [reply]
Glad to hear you got it working.  If this is solved, can you mark the discussion as solved?

**Reply by vlad2005** [reply]
Yes, but how?
I don't see any button, menu or something like that to mark post as solved.

**Reply by quasipickle** [reply]
I think you can mark a comment as the answer: https://docs.github.com/en/discussions/managing-discussions-for-your-community/moderating-discussions
